src/Microsoft.DotNet.Core.Build.Tasks/LockFileCache.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Concurrent;
+using NuGet.Common;
+using NuGet.ProjectModel;
+
+namespace Microsoft.DotNet.Cli.Tasks
+{
+    internal class LockFileCache
+    {
+        public static LockFileCache Instance { get; } = new LockFileCache();
```

Comment:
Not very happy that MSBuild does not have a dependency injection story for tasks, which forces us to rely on things like this to compose our class hierarchies.